### PR TITLE
enh(UI): Refactor snackbar to integrate severity

### PR DESCRIPTION
**Pull request by Thebarda**

### CODE REVIEW

File: src/Snackbar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    opacity: 0.9,
+  },
+  message: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+}));
+
+interface Props {
+  message?: string;
```
Comment (kduret) [inline src/Snackbar/index.tsx]: I think this prop should be mandatory

File: src/Snackbar/useMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+              {err}
+            </p>
+          ))}
+        </div>
+      ),
+      newSeverity,
+    });
+  };
+
+  console.log(severity);
```
Comment (kduret) [inline src/Snackbar/useMessage.tsx]: Remove debug

File: src/Snackbar/withSnackbar.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          onClose={confirmMessage}
+          open={hasMessage}
+          message={message}
+        />
+      </Context.Provider>
+    );
+  };
+};
+
+export default withSnackbar;
```
Comment (kduret) [inline src/Snackbar/withSnackbar.tsx]: Prefer to put default export as last one

File: src/Snackbar/withSnackbar.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React, { createContext, ReactNode, ReactElement } from 'react';
+
+import Snackbar from '.';
+import useMessage from './useMessage';
+
+export interface SnackbarState {
+  showMessage: ({ newMessage, newSeverity }) => void;
+  showMessages: ({ newMessages, newSeverity }) => void;
```
Comment (bdauria) [inline src/Snackbar/withSnackbar.tsx]:
```suggestion
  showMessage: ({ message, severity }) => void;
  showMessages: ({ messages, severity }) => void;
```

File: src/Snackbar/useMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { SnackbarState } from './withSnackbar';
+
+interface UseMessage extends SnackbarState {
```
Comment (bdauria) [inline src/Snackbar/useMessage.tsx]:
```suggestion
interface SnackbarContent {
```

File: src/Snackbar/useMessage.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  message;
+  severity;
+  confirmMessage: () => void;
+}
+
+const useMessage = (): UseMessage => {
```
Comment (bdauria) [inline src/Snackbar/useMessage.tsx]:
```suggestion
const useMessage = (): SnackbarContent & SnackbarActions => {
```

File: src/Snackbar/withSnackbar.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
+import React, { createContext, ReactNode, ReactElement } from 'react';
+
+import Snackbar from '.';
+import useMessage from './useMessage';
+
+export interface SnackbarState {
```
Comment (bdauria) [inline src/Snackbar/withSnackbar.tsx]:
```suggestion
export interface SnackbarActions {
```

File: src/Snackbar/useMessage.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { SnackbarState } from './withSnackbar';
```
Comment (bdauria) [inline src/Snackbar/useMessage.tsx]:
```suggestion
import { SnackbarActions } from './withSnackbar';
```

File: src/Snackbar/withSnackbar.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  showMessage: ({ newMessage, newSeverity }) => void;
+  showMessages: ({ newMessages, newSeverity }) => void;
+}
+
+const noOp = (): void => undefined;
+
+const defaultSnackBarState: SnackbarState = {
```
Comment (bdauria) [inline src/Snackbar/withSnackbar.tsx]:
```suggestion
const defaultSnackBarActions: SnackbarActions = {
```

File: src/Snackbar/withSnackbar.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  showMessage: noOp,
+  showMessages: noOp,
+};
+
+const Context = createContext<SnackbarState>(defaultSnackBarState);
```
Comment (bdauria) [inline src/Snackbar/withSnackbar.tsx]:
```suggestion
const Context = createContext<SnackbarActions>(defaultSnackBarActions);
```

File: src/Snackbar/index.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+export const errorSnackbar = (): JSX.Element => (
+  <Snackbar
+    open
+    message="Something unexpected happened..."
+    severity={Severity.error}
+  />
+);
+
+export const normal = (): JSX.Element => (
```
Comment (jdelpierre) [inline src/Snackbar/index.stories.tsx]: Why did you choose "normal"? 
Isn't "default" more common?

File: src/Snackbar/useMessage.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+      [],
+    );
+
+    showMessage({
+      message: (
+        <div style={{ display: 'block' }}>
+          {formattedMessages.map((err, index) => (
+            <p style={{ margin: 0 }} key={messageKeys[index]}>
+              {err}
+            </p>
```
Comment (kduret) [inline src/Snackbar/useMessage.tsx]:
```suggestion
          {formattedMessages.map((errorMessage, index) => (
            <p style={{ margin: 0 }} key={messageKeys[index]}>
              {errorMessage}
            </p>
```

File: src/Snackbar/index.stories.tsx (hunk 11 above)
Comment (Thebarda) [reply src/Snackbar/index.stories.tsx]: You're right but default is a reserved word

Comment (Thebarda) [reply src/Snackbar/index.stories.tsx]: Severity is optional thus, when it is undefined, default severity is success

File: src/Snackbar/index.tsx [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+  info: {
+    Icon: (props): JSX.Element => <Info {...props} />,
+    getColor: (theme): string => theme.palette.info.dark,
+  },
+};
+
+const ErrorSnackbar = ({
+  message,
+  open,
+  onClose = undefined,
```
Comment (bdauria) [inline src/Snackbar/index.tsx]:
```suggestion
  onClose,
```

**Pull request merged**